services/deduplicate/index.ts

```diff
@@ -96,26 +96,28 @@ export function mainDedupe() {
   const blocks: Block[] = select('core/block-editor').getBlocks();
   const {
     wp_curate_deduplication: wpCurateDeduplication = true,
+    wp_curate_unique_pinned_posts: wpCurateUniquePinnedPosts = false,
     // @ts-ignore
   } = select('core/editor').getEditedPostAttribute('meta') || {};
 
   const queryBlocks: Block[] = [];
   // Loop through all blocks and find all query blocks.
   getQueryBlocks(blocks, queryBlocks);
-  // Loop through all query blocks and find all pinned posts. Add them to the list of used ids.
-  // Note: This means that a pinned post will be used only once on the page. It cannot be backfilled
-  // earlier on the page. The front end does not currently do this, so it will be commented out
-  // here to match. We should probably add a setting to allow this in both places.
-
-  // queryBlocks.forEach((queryBlock) => {
-  //   const { attributes } = queryBlock;
-  //   const { posts } = attributes;
-  //   posts?.forEach((post) => {
-  //     if (post) {
-  //       deduplicate(post);
-  //     }
-  //   });
-  // });
+
+  /**
+   * This block of code is responsible for enforcing the unique pinned posts setting in the editor.
+   */
+  if (wpCurateUniquePinnedPosts) {
+    queryBlocks.forEach((queryBlock) => {
+      const { attributes } = queryBlock;
+      const { posts } = attributes;
+      posts?.forEach((post) => {
```

Comment:
We can collapse this quite a bit:
```suggestion
    queryBlocks.forEach((queryBlock) => {
      queryBlock?.attributes?.posts?.forEach((post) => {
```